content/chat/rooms/message-reactions.textile

```diff
@@ -0,0 +1,192 @@
+
+Always call @Message.with(event)@ when applying message events and reaction events to existing messages to ensure that reactions are correctly copied or updated. **Do not** replace existing messages with messages received from events as reactions will be lost.
+
+
+h2(#subscribe). Subscribing to message reactions
+
+Ably generates a summary (aggregate) of the reactions for each message and for each reaction type. For displaying accurate counts for message reactions, subscribe to changes in the message summary.
+
+```[javascript]
+    room.messages.reactions.subscribe((event) => {
+        console.log("received reactions summary event", event);
+    });
+```
+
+The event is of type @reaction.summary@. @event.summary@ is the received reactions summary and contains the following properties:
+
+|_. Property |_. Description |_. Example |
+| @messageSerial@ | Serial of the message this summary is for. | @01826232498871-001&#64;abcdefghij:001@ |
+| @timestamp@ | Timestamp of the summary. | @new Date("2025-04-22T11:37:59.988Z")@ |
+| @unique@ | Unique reactions summary. | @{ "👍": { total: 2, clientIds: ["a", "b"]} }@ |
+| @distinct@ | Distinct reactions summary. | @{ "👍": { total: 2, clientIds: ["a", "b"]} }@ |
+| @multiple@ | Multiple reactions summary. | @{ "👍": { total: 5, clientIds: {"a": 2, "b": 3} }@ |
+
+Message reaction summary events can be used with @Message.with(event)@ to get an updated message object, with the reactions applied correctly. Similarly, when calling @Message.with()@ with other message events (for example @MessageEvents.Updated@), the reactions summary will be correctly preserved in the resulting message object.
+
+Example usage:
+
+```[javascript]
+    // init messages, in practice this should be updated with a message subscription
+    let messages = await room.messages.get({limit: 50});
+
+    // subscribe to message reactions summary events
+    room.messages.reactions.subscribe((event) => {
+        // find the relevant message (in practice: use binary search or a map for lookups)
+        const idx = messages.findLastIndex((msg) => msg.serial === event.messageSerial);
+        if (idx === -1) {
+            // not found
+            return;
+        }
+        // update message
+        messages[idx] = messages[idx].with(event);
+    });
+```
+
+h3(#throttle). Summary events are sent efficiently at scale
+
+Summary events are typically created and published immediately after a reaction is added or removed. If the reaction is a no-op (for example, when removing a reaction that didn't exist), then there will be no summary event.
+
+If multiple reactions are added in a short period of time, multiple reactions may be rolled up and only a single summary event will be published that contains the aggregated results of all reactions. This reduces Ably platform costs by reducing the number of messages in busy rooms.
```

Comment:
```suggestion
If multiple reactions are added in a short period of time, multiple reactions may be rolled up and only a single summary event will be published that contains the aggregated results of all reactions. This reduces the number of outbound messages and thus your costs in busy rooms.
```

(The previous version sounded a bit more like it was saving us money but not the customer!)

Reply:
done